### DISCUSSION TITLE
Reverse a string

### DISCUSSION THREAD

**Discussion by aditii144** [aditii144/c-#2] [Q&A] [answered]
how to reverse a string in c programming?

**Comment by ujjwalbhandarii** [comment, marked as answer]
Easy method would be by using strrev() function.
It is a built-in function in C and is defined in string.h header file. The strrev() function is used to reverse the given string.

C code showcasing strrev function :

#include <stdio.h>
#include <string.h>
int main()
{
char s[100];
printf("Enter a string to reverse\n");
gets(s);
strrev(s);
printf("Reverse of the string: %s\n", s);
return 0;
}

Also there is alternative way :

#include <stdio.h>
int main()
{
char s[1000], r[1000];
int begin, end, count = 0;
printf("Input a string\n");
gets(s);
// Calculating string length
while (s[count] != '\0')
count++;
end = count - 1;
for (begin = 0; begin < count; begin++) {
r[begin] = s[end];
end--;
}
r[begin] = '\0';
printf("%s\n", r);
return 0;
}

**Comment by aditii144** [comment]
thankyou! this is helpful. also, can we reverse a string using recursion?

**Comment by ujjwalbhandarii** [comment]
// reversing a string using recursion 

#include <stdio.h>
#include <string.h>
void reverse(char*, int, int);

int main()
{
   char a[100];
   gets(a);
   reverse(a, 0, strlen(a)-1);
   printf("%s\n", a);
   return 0;
}

void reverse(char *x, int begin, int end)
{
   char c;

   if (begin >= end)
      return;

   c          = *(x+begin);
   *(x+begin) = *(x+end);
   *(x+end)   = c;

   reverse(x, ++begin, --end);
}
